From 7490e1dbb02fc487415156ff6c6d1e14d570c1e7 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Thu, 2 Jan 2020 17:32:58 +0800
Subject: [PATCH] files: add lable if /mnt /opt /media is symlinks file

if /mnt, /opt, /media is symlink files they will have a default
label, add rules to label these symlink files.

Upstream-Status: Pending

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 policy/modules/kernel/files.fc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/kernel/files.fc b/policy/modules/kernel/files.fc
index f69900945..5fff8b92c 100644
--- a/policy/modules/kernel/files.fc
+++ b/policy/modules/kernel/files.fc
@@ -116,6 +116,7 @@ HOME_ROOT/lost\+found/.*	<<none>>
 #
 # Mount points; do not relabel subdirectories, since
 # we don't want to change any removable media by default.
+/media			-l	gen_context(system_u:object_r:mnt_t,s0)
 /media(/[^/]*)		-l	gen_context(system_u:object_r:mnt_t,s0)
 /media(/[^/]*)?		-d	gen_context(system_u:object_r:mnt_t,s0)
 /media/[^/]*/.*			<<none>>
@@ -129,6 +130,7 @@ HOME_ROOT/lost\+found/.*	<<none>>
 #
 # /mnt
 #
+/mnt			-l	gen_context(system_u:object_r:mnt_t,s0)
 /mnt(/[^/]*)		-l	gen_context(system_u:object_r:mnt_t,s0)
 /mnt(/[^/]*)?		-d	gen_context(system_u:object_r:mnt_t,s0)
 /mnt/[^/]*/.*			<<none>>
@@ -142,6 +144,7 @@ HOME_ROOT/lost\+found/.*	<<none>>
 # /opt
 #
 /opt			-d	gen_context(system_u:object_r:usr_t,s0)
+/opt			-l	gen_context(system_u:object_r:usr_t,s0)
 /opt/.*				gen_context(system_u:object_r:usr_t,s0)
 
 /opt/(.*/)?var/lib(64)?(/.*)?	gen_context(system_u:object_r:var_lib_t,s0)
-- 
2.17.1

